app/views/onboarding/OnboardingPermissions.js

```diff
@@ -55,73 +43,48 @@ export const OnboardingPermissions = ({ navigation }) => {
     moveToNextStep();
   };
 
-  const handleOnPressDone = () => {
-    SetStoreData(PARTICIPATE, location.status === PermissionStatus.GRANTED);
-    SetStoreData(ONBOARDING_DONE, true);
-    navigation.replace('Main');
-  };
-
-  const locationStep = {
-    titleText: languages.t('label.launch_enable_location'),
-    subTitleText: languages.t('label.launch_subheader'),
-    indicatorText: languages.t('label.launch_access_location'),
-    status: location.status,
-    buttonLabel: languages.t('label.launch_enable_location'),
-    handleButtonPress: handleRequestLocation,
-    testID: 'location-indicator',
-  };
   const notificationStep = {
-    titleText: languages.t('label.launch_notif_header'),
-    subTitleText: languages.t('label.launch_notif_subheader'),
-    indicatorText: languages.t('label.launch_notification_access'),
-    status: notification.status,
-    buttonLabel: languages.t('label.launch_enable_notif'),
+    header: languages.t('onboarding.notification_header'),
+    subHeader: languages.t('onboarding.notification_subheader'),
+    nextStepParam: 'location',
     handleButtonPress: handleRequestNotifications,
-    testID: 'notification-indicator',
-  };
-  const authSubscriptionStep = {
-    titleText: languages.t('label.launch_authority_header'),
-    subTitleText: languages.t('label.launch_authority_subheader'),
-    indicatorText: languages.t('label.launch_authority_access'),
-    status: authSubscription.status,
-    buttonLabel: languages.t('label.launch_enable_auto_subscription'),
-    handleButtonPress: handleRequestAuthSubscription,
-    testID: 'auth-subscription-indicator',
-  };
-  const doneStep = {
-    titleText: languages.t('label.launch_done_header'),
-    subTitleText: languages.t('label.launch_done_subheader'),
-    buttonLabel: languages.t('label.launch_finish_set_up'),
-    handleButtonPress: handleOnPressDone,
-  };
+    icon: Icons.Bell,
+    buttonLabel: languages.t('label.launch_enable_notif')
+  }
+  const locationStep = {
+    header: languages.t('onboarding.location_header'),
+    subHeader: languages.t('onboarding.location_subheader'),
+    nextStepParam: null,
+    handleButtonPress: handleRequestLocation,
+    icon: Icons.LocationPin,
+    buttonLabel: languages.t('label.launch_allow_location')
+  }
+  const steps = {
+    notification: notificationStep,
+    location: locationStep,
+  }
+  const currentStep = steps[route.params.step]
```

Comment:
ah I see the idea now. Yeah this totally works! Though in general i try to avoid relying on the navigation components for handling application data/logic, especially for mobile apps, as it increases the coupling between the screens making it harder to update flows in the future.  And its highly likely that we'll be updating theses flows as the product evolves.

What do you think about having the screen component itself determine what step it should be on?

Reply:
That certainly is an option.

The reason I've built it this way is so there's a clear transition between the request notification permission screen and the request location permission screen using the stack navigator.

Reply:
I agree that this screen will probably be updated often in the future so I can look into changing it as you described.